MdeModulePkg/Core/DxeIplPeim/DxeHandoff.c

```diff
@@ -38,26 +38,26 @@ HandOffToDxeCore (
   //
   BaseOfStack = AllocatePages (EFI_SIZE_TO_PAGES (STACK_SIZE));
   ASSERT (BaseOfStack != NULL);
+  // MU_CHANGE Start Always set NX for stack
+  // if (PcdGetBool (PcdSetNxForStack)) {
+  Status = PeiServicesLocatePpi (
+             &gEdkiiMemoryAttributePpiGuid,
+             0,
+             NULL,
+             (VOID **)&MemoryPpi
+             );
+  ASSERT_EFI_ERROR (Status);

```

Comment:
Should do more than assert here to prevent a potentially invalid dereference below.